### PR TITLE
Basename regex fix

### PR DESCRIPTION
**Pull request by Shaolans**
Please make sure the below checklist is followed for Pull Requests.
  - [x] [Travis tests](https://travis-ci.org/jhipster/jhipster-umk/pull_requests) are green
  - [x] Tests are added where necessary
  - [x] Documentation is added/updated where necessary
  - [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-core/blob/master/CONTRIBUTING.md) are followed

Fix https://github.com/jhipster/generator-jhipster/issues/9699

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Is there a unit test to add or update ?

**Comment by Shaolans** [comment]
I added a simple test to check this case.

File: test/spec/jdl/jdl_importer_test.js [hunk 1]
```diff
@@ -1003,6 +1003,26 @@ describe('JDLImporter', () => {
         expect(returned.exportedEntities[0].fields[0].fieldValidateRulesPattern.includes("\\'")).to.be.true;
       });
     });
+    context('when parseing a JDL with underscores contained in the application name', () => {
```
Comment (MathieuAA) [inline test/spec/jdl/jdl_importer_test.js]: parsing

**Comment by MathieuAA** [comment]
Thanks for this :) I'll merge this once the type is fixed

**Pull request merged**